smartReceiptsLibrary/src/main/java/co/smartreceipts/android/workers/reports/pdf/pdfbox/DefaultPdfBoxContext.java

```
@@ -0,0 +1,105 @@
+package co.smartreceipts.android.workers.reports.pdf.pdfbox;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+
+import com.tom_roush.pdfbox.pdmodel.common.PDRectangle;
+import com.tom_roush.pdfbox.util.awt.AWTColor;
+
+import java.util.Map;
+
+import co.smartreceipts.android.persistence.Preferences;
+
+public class DefaultPdfBoxContext implements PdfBoxContext {
+
+    /**
+     * TODO
+     * move all these to {@link PdfBoxContext} or to the appropriate section??
+     */
+    public static final String FONT_DEFAULT = "FONT_DEFAULT";
+    public static final String FONT_TABLE_HEADER = "FONT_TABLE_HEADER";
+    public static final String FONT_SMALL = "FONT_SMALL";
+    public static final String FONT_TITLE = "FONT_TITLE";
+    public static final String COLOR_DARK_BLUE = "DARK_BLUE";
+    public static final String COLOR_HEADER = "HEADER";
+    public static final String COLOR_CELL = "CELL";
+
+
+    private final Context mContext;
+    private final Preferences mPreferences;
+
+    private Map<String, AWTColor> mColors;
+    private Map<String, FontSpec> mFonts;
+    private PDRectangle mPageSize = PDRectangle.A4;
+
+    public DefaultPdfBoxContext(@NonNull Context context,
+                                @NonNull Preferences preferences) {
+        mContext = context;
+        mPreferences = preferences;
```

Minor item (and nbd to change for this review but for the future), could you wrap all the @NonNull items with a preconditions check, so we can fail fast:

mPrefs = Preconditions.checkNotNull(prefs);